exchange-deployer/src/deploy_types/common.rs

```diff
@@ -0,0 +1,27 @@
+multiversx_sc::imports!();
+
+pub const GAS_AFTER_ASYNC: u64 = 10_000;
+
+#[multiversx_sc::module]
+pub trait CommonModule {
+    fn get_default_code_metadata(&self) -> CodeMetadata {
+        CodeMetadata::PAYABLE_BY_SC | CodeMetadata::READABLE | CodeMetadata::UPGRADEABLE
+    }
+
+    fn require_deployed_contract(
+        &self,
+        user_address: &ManagedAddress,
+        sc_address: &ManagedAddress,
+    ) {
+        require!(
+            self.deployed_contracts(user_address).contains(sc_address),
+            "Cannot perform action, contract deployed by another account"
+        );
+    }
+
+    #[storage_mapper("deployedContracts")]
+    fn deployed_contracts(
```

Comment:
Indeed, from what I saw from the general ProxyDeployer SC, having the information about all the deployed contracts of a specific type would be very helpful (for the microservice). Maybe add a new storage, as Costin suggested, having all the deployed contracts under a contract type key.